src/index.js

```diff
@@ -1,5 +1,61 @@
 'use strict';
 
-function nearestVector() {}
+const squaredDistance = require('ml-distance-euclidean').squared;
 
-module.exports = nearestVector();
\ No newline at end of file
+const defaultOptions = {
+    distanceFunction: squaredDistance,
+    similarityFunction: false,
+    returnObject: false
+};
+
+/**
+ *
+ * @param {Array<Array<Number>>} listVectors - List of vectors with same dimensions
+ * @param {Array<Number>} vector - Reference vector to "classify"
+ * @param {Object} [options] - Options object
+ * @param {Function} [options.distanceFunction = squaredDistance] - Function that receives two vectors and return their distance value as number
+ * @param {Function} [options.similarityFunction = undefined] - Function that receives two vectors and return their similarity value as number
+ * @param {Boolean} [options.returnObject = false] - Return the nearest vector instead of the index of it
+ * @return {Number|Array<Number>} - The index or the content of the nearest vector
+ */
+function nearestVector(listVectors, vector, options) {
+    options = options || defaultOptions;
+    options.distanceFunction = options.distanceFunction || defaultOptions.distanceFunction;
+    options.similarityFunction = options.similarityFunction || defaultOptions.similarityFunction;
+    options.returnObject = options.returnObject || defaultOptions.returnObject;
+
+    var vectorIndex = -1;
+    if (typeof options.similarityFunction === 'function') {
+
+        // maximum similarity
+        var maxSim = -1;
```

Comment:
A similarity should be defined in the set [0, 1], but It's not a problem to change that